lib/app_manifest/serializer.rb

```diff
@@ -0,0 +1,34 @@
+module AppManifest
+  # Serialization logic for AppManifest models
+  module Serializer
```

Comment:
I ran into a bug while testing this in staging. If I have a manifest like this:

```
{
  "environments": {
    "test": {
      "addons": []
    }
  }
}
```

then the `environments.test.addons` array isn't preserved. This causes a bug when we try to explicitly override the regular `addons` hash with an empty array for the test env.